simple_use_case/pydoit/dodo.py

```diff
@@ -53,28 +61,67 @@ def task_plot_over_line():
     pol = ROOT / "plotoverline.csv"
     return {
         "file_dep": [pvdfile, vtufile, postproc],
-        "actions": [f"pvbatch {postproc} {pvdfile} {pol}"],
+        "actions": [f"pvbatch {postproc} -i {pvdfile} -o {pol}"],
         "targets": [pol],
         "clean": True,
     }
 
 
-def task_copy_paper():
-    """copy paper.tex from source directory"""
-    source_code = SOURCE / "paper.tex"
-    root_code = ROOT / "paper.tex"
+def task_write_paper_source():
+    """write paper source to file"""
+
+    def write_source(gmsh_stdout, dependencies, targets):
+        # file names in correct order
+        fnames = ["paper.tex", "lineplot.tex", "plotoverline.csv", "numdofs.txt"]
+        deps = fix_deps(dependencies, fnames)
+
+        with open(deps[0], "r") as infile:
+            paper = infile.read()
+
+        # parse stdout from task_generate_mesh to determine domain size
+        s = gmsh_stdout.split("Used domain size:")[1]
+        size = s.split("Used mesh size")[0]
+
+        # read file written by task_poisson to determine number of DoFs
+        with open(deps[3], "r") as handle:
+            ndofs = handle.read()
+
+        plot = "\\inputplot{{{}}}{{{}}}{{{}}}{{{}}}\n".format(deps[1], deps[2], float(size), int(ndofs))
+        paper = paper.replace("% TODO inputplot", plot)
+
+        with open(targets[0], "w") as outfile:
+            outfile.write(paper)
```

Comment:
While the use of the python-action for this task is natural in `doit`, what bothers me is that using small snippets we end up with the same code snippet (in whatever programming language the tool supports) for each tool (code duplication). We could avoid this by putting the contents of `write_source` in a separate script, but this feels a bit like overkill (with a proper CLI for all the arguments). Also it makes the definition of the `task_write_paper_source` unnecesarry complicated (in the case of `doit`).
Which option do you prefer @joergfunger and @dglaeser ?